File: Leetcode/Python/_1189.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        b = text.count('b')
        a = text.count('a')
        l = int(text.count('l') / 2)
        o = int(text.count('o') / 2)
        n = text.count('n')
        f = [b, a, l, o, n]
        return min(f)
      
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        hashMap = collections.Counter(text)
        b = hashMap.get('b', 0)
        a = hashMap.get('a', 0)
        l = int(hashMap.get('l', 0)/2)
        o = int(hashMap.get('o', 0)/2)
        n = hashMap.get('n', 0)
        f = [b, a, l, o, n]
        return min(f)
      
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        hashMap = {"b": 1, "a": 1, 
                   "l": 2, "o": 2, "n": 1}
        currentMap = {}
        for letter in text:
            currentMap[letter] = currentMap.get(letter, 0) + 1
        count = inf
        for key in hashMap.keys():
            count = min(count, currentMap.get(key, 0) // hashMap[key])
        return count

File: Leetcode/cpp/1189.cpp
#include <iostream>
#include <climits>
#include <string>
#include <cassert>
#include <unordered_map>

class Solution {
public:
    int maxNumberOfBalloons(std::string text) {
        std::unordered_map<char, int> umap;
        for (auto chr : text) {
            umap[chr] += 1;
        }
        std::string word = "balloon";
        int minLetters = INT_MAX;
        int counter = 0;
        for (auto chr : word) {
            if (!umap.contains(chr)) {
                return 0;
            } else {
                counter = umap[chr];
                if (chr == 'l' || chr == 'o') {
                    counter /= 2;
                }
            }
            minLetters = std::min<int>(minLetters, counter);
        }
        return minLetters;
    }
};

int main() {
    Solution sol;
    std::string text = "loonbalxballpoon";
    int expectedOutput = 2;
    int output = sol.maxNumberOfBalloons(text);
    
    assert((output == expectedOutput) && "Test #1 failed!");
    std::cout << "Tests are passed!" << std::endl;
    return 0;
}
